src/containers/app/components/TxForm.jsx

```diff
@@ -35,9 +35,13 @@ class TxForm extends React.Component {
   }
 
   executeExchange = async () => {
-    this.props.submitDeposit()
-    const txHash = await executeDeposit(this.props.address, this.state.amount, this.props.networkID)
-    this.props.updateDepositHash(txHash)
+    try {     
+      const txHash = await executeDeposit(this.props.address, this.state.amount, this.props.networkID)
+      this.props.submitDeposit()
```

Comment:
`submitDeposit` needs to happen before `executeDeposit` because it causes the UI to enter the loading state, otherwise it will only change once the deposit is submitted

Reply:
@ansermino the way it was set up after the refactor froze the application on errors with respect to the executeDeposit call. If you call the submitDeposit action first it will trigger the loading screen. If you know of a better way to handle this issue, otherwise we'd have to rewrite the logic throughout the app.